### PR TITLE
issue 2050 ifNoneMatch returns 412 unless ifNoneMatchReturnsNotModified

### PR DESCRIPTION
**Pull request by punktilious**
Signed-off-by: Robin Arnold <robin.arnold@ibm.com>

### CODE REVIEW

File: fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorPersist.java [hunk 1]
```diff
@@ -300,6 +302,21 @@ private void doInteraction(int entryIndex, String requestDescription, long initi
                         .build())
                     .build();
             setEntryComplete(entryIndex, entry, requestDescription, initialTime);
+        } catch (FHIRPersistenceIfNoneMatchException e) {
+            if (failFast) {
+                String msg = "Error while processing request bundle.";
+                throw new FHIRRestBundledRequestException(msg, e).withIssue(e.getIssues());
+            }
```
Comment (lmsurpre) [inline fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorPersist.java:309]: I updated some very similar code in the past to include the entryIndex in both the error message and the issue...maybe we should do the same in each catch block here?

Comment (lmsurpre) [reply fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorPersist.java:309]: https://github.com/IBM/FHIR/pull/2960/files#diff-35b7e53e0ed72c5da33f937ee4634c89bb018a496a8c2db4e220a75117c59507R396

Comment (punktilious) [reply fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorPersist.java:309]: Yep, I'll do that

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**